socorro/signature/rules.py

```diff
@@ -725,12 +725,20 @@ def action(self, raw_crash, processed_crash, notes):
             new_sig = 'IPCError-browser | {}'
         else:
             new_sig = 'IPCError-content | {}'
-        new_sig = new_sig.format(raw_crash['ipc_channel_error'][:100])
 
-        notes.append(
-            'Signature replaced with an IPC Channel Error, '
-            'was: "{}"'.format(processed_crash['signature'])
-        )
+        try:
+            new_sig = new_sig.format(raw_crash['ipc_channel_error'][:100])
+        except UnicodeEncodeError as ex:
+            new_sig = new_sig.format(ex)
```

Comment:
Mmm... We don't want the exception to be part of the signature. That doesn't make any sense in regards to what signatures are.

I think instead of doing what you're doing here, I'd do what we do with the `AbortSignature` rule where it drops non-ascii characters using `socorro.lib.util.drop_unicode` (which has both a terrible name and a terrible docstring) and then goes with that.

Reply:
Ok! I'll try.